src/js/fluxcontroller.js

```diff
@@ -218,16 +363,9 @@ define(function (require, exports, module) {
                         throw new Error("Transfer passed an undefined action");
                     }
 
-                    var nextReads = _.union(nextAction.reads, nextAction.writes) || locks.ALL_LOCKS;
-                    if (!_subseteq(nextReads, currentReads)) {
-                        log.error("Missing read locks:", _.difference(nextReads, currentReads).join(", "));
-                        throw new Error("Next action requires additional read locks");
-                    }
-
-                    var nextWrites = nextAction.writes || locks.ALL_LOCKS;
-                    if (!_subseteq(nextWrites, currentWrites)) {
-                        log.error("Missing write locks:", _.difference(nextWrites, currentWrites).join(", "));
-                        throw new Error("Next action requires additional write locks");
+                    if (!currentTransfers.has(nextAction)) {
+                        var nextActionName = self._actionNames.get(nextAction);
+                        throw new Error("Invalid transfer from " + actionName + " to " + nextActionName);
```

Comment:
You should probably have a more descriptive error here telling the dev they forgot to declare a transfer.